packages/server-core/src/media/static-resource/static-resource.hooks.ts

```diff
@@ -80,20 +83,42 @@ export default {
       schemaHooks.resolveQuery(staticResourceQueryResolver)
     ],
     find: [
-      iff(isProvider('external'), verifyScope('static_resource', 'read')),
+      iff(
+        isProvider('external'),
+        iffElse(
+          checkScope('static_resource', 'read'),
+          [],
+          [verifyScope('editor', 'write'), resolveProjectId(), verifyProjectPermission(['owner', 'editor', 'reviewer'])]
+        )
+      ),
       discardQuery('action'),
+      discardQuery('projectId'),
       collectAnalytics()
     ],
     get: [disallow('external')],
     create: [
-      iff(isProvider('external'), verifyScope('static_resource', 'write')),
+      iff(
+        isProvider('external'),
+        iffElse(
+          checkScope('static_resource', 'write'),
+          [],
+          [verifyScope('editor', 'write'), resolveProjectId(), verifyProjectPermission(['owner', 'editor'])]
+        )
+      ),
       setLoggedinUserInBody('userId'),
       () => schemaHooks.validateData(staticResourceDataValidator),
       schemaHooks.resolveData(staticResourceDataResolver)
     ],
     update: [disallow()],
     patch: [
-      iff(isProvider('external'), verifyScope('static_resource', 'write')),
+      iff(
```

Comment:
I think we need a similar check for remove hook as well? on line 125 below

Reply:
checks have been added for remove too